openlibrary/plugins/worksearch/code.py

```diff
@@ -642,12 +642,20 @@ def works_by_author(akey, sort='editions', page=1, rows=100, has_fulltext=False,
         'first_publish_year', 'public_scan_b', 'lending_edition_s', 'lending_identifier_s',
         'ia_collection_s', 'cover_i']
     fl = ','.join(fields)
-    fq = ['author_key:' + akey, 'type:work']
+    fq = 'author_key:' + akey
     if has_fulltext:
         fq.append('has_fulltext:true')
-    solr_select = solr_select_url + "?fq=%s&q.op=AND&q=%s&start=%d&rows=%d&fl=%s&wt=json" % ('&fq='.join(fq), q, offset, rows, fl)
+    params = {
+        'fq': fq,
+        'q': q,
+        'start': offset,
+        'rows': rows,
+        'fl': fl,
+    }
+    solr_select = solr_select_url + "?fq=type:work&wt=json&q.op=AND&"
+    solr_select = solr_select_url + urllib.parse.urlencode(params, 'utf-8')
```

Comment:
I think this is actually wrong -- we're clobbering `solr_select` and shouldn't be. We should be saying 
`solr_select += urllib.parse.urlencode(params, 'utf-8')`

Reply:
`solr_select` and `solr_select_url` are two separate things so the += would not be right however I do believe that line 655 could safely be deleted.